src/kmeans.jl

```diff
@@ -154,25 +181,19 @@ function _kmeans!(
         end
     end
 
-    return KmeansResult(convert(Matrix, centers), assignments, costs, counts,
-                        cweights, Float64(objv), t, converged)
+    return KmeansResult(centers, assignments, costs, counts,
+                        cweights, objv, t, converged)
 end
 
+####
+#### update assignments
+####
 
-#
-#  Updates assignments, costs, and counts based on
-#  an updated (squared) distance matrix
-#
-function update_assignments!(
-    dmat::Matrix{T},              # in:  distance matrix (k x n)
-    is_init::Bool,                # in:  whether it is the initial run
-    assignments::Vector{Int},     # out: assignment vector (n)
-    costs::Vector{T},             # out: costs of the resultant assignment (n)
-    counts::Vector{Int},          # out: # samples assigned to each cluster (k)
-    to_update::Vector{Bool},      # out: whether a center needs update (k)
-    unused::Vector{Int}          # out: list of centers with no samples
-    ) where T<:Real
-
+function update_assignments!(dmat::Matrix{T}, is_init::Bool,
```

Comment:
The old one was more readable. It's too bad to discard detailed comments. Same elsewhere.

Reply:
ok